### PR TITLE
Handling review request removed

### PR DESCRIPTION
**Pull request by hdamico**
## Description

Now we can handle `review_request_removed` requests from github.

### CODE REVIEW

File: app/services/github_handler.rb [hunk 1]
```diff
@@ -45,6 +45,14 @@ def handle_review_request
     pr.review_requests.create!(data: payload, owner: owner, reviewer: reviewer)
   end
 
+  def handle_review_removal
+    pr = PullRequest.find_by!(github_id: payload['pull_request']['id'])
+    pr.transaction do
+      pr.review_requests = []
```
Comment (santib) [inline app/services/github_handler.rb]: why is this needed? Isn't `dependent: :destroy` here https://github.com/rootstrap/rs-code-review-metrics/blob/master/app/models/pull_request.rb#L26 going to take care of this already?

Comment (hdamico) [reply app/services/github_handler.rb]: You are right! Committing the changes

File: app/services/github_handler.rb [hunk 2]
```diff
@@ -45,6 +45,10 @@ def handle_review_request
     pr.review_requests.create!(data: payload, owner: owner, reviewer: reviewer)
   end
 
+  def handle_review_removal
+    PullRequest.find_by!(github_id: payload['pull_request']['id']).destroy!
```
Comment (santib) [inline app/services/github_handler.rb]: Maybe I'm missing something, but why removing a review request destroys a `PullRequest`? I thought we would be just destroying the `ReviewRequest`

Comment (hdamico) [reply app/services/github_handler.rb]: Yeah, i got a miss understanding. Thanks!

File: app/services/github_handler.rb [hunk 3]
```diff
@@ -45,6 +45,12 @@ def handle_review_request
     pr.review_requests.create!(data: payload, owner: owner, reviewer: reviewer)
   end
 
+  def handle_review_removal
+    reviewer = User.find_by(github_id: payload['requested_reviewer']['id'])
+    pr = PullRequest.find_by(github_id: payload['pull_request']['id'])
+    pr.review_requests.find_by!(reviewer: reviewer).destroy!
```
Comment (santib) [inline app/services/github_handler.rb]: do we care about an exception if the review_request isn't found?

maybe we can do `pr.review_requests.find_by(reviewer: reviewer)&.destroy!`

thougths?

File: spec/services/github_handler_spec.rb [hunk 4]
```diff
@@ -81,5 +81,22 @@
...
+        create :user,
+          github_id: 1001,
+          login: 'pentacat',
+          node_id: 'MDExOlB1bGxc5MTQ3NDM3'
+      }
+      let!(:review_requests) { ReviewRequest.create(
+                                pull_request: pull_request, owner: user, reviewer: user
+                              ) }
+
+      it 'deletes review requests' do
```
Comment (santib) [inline spec/services/github_handler_spec.rb]: minor comment, `deleted review request` in singular

File: app/services/github_handler.rb (hunk 3 above)
Comment (santiagovidal) [reply app/services/github_handler.rb]: Do we want to destroy these requests? Or mark them as invalid?

For example, if person A sends a review request to person B and then removes it after 3 days, do we want that review request to affect the statistics?

I think so but what does everyone think?

@anthfb @matiasmansilla1989 @santib

File: spec/services/github_handler_spec.rb [hunk 5]
```diff
@@ -81,5 +81,22 @@
...
+
+    describe 'review removal request' do
+      let!(:pull_request) { create :pull_request, github_id: 1000 }
+      let!(:user) {
+        create :user,
+          github_id: 1001,
+          login: 'pentacat',
+          node_id: 'MDExOlB1bGxc5MTQ3NDM3'
+      }
+      let!(:review_requests) { ReviewRequest.create(
```
Comment (santiagovidal) [inline spec/services/github_handler_spec.rb]: `do ... end` blocks are preferred for multi-line `let` variables. Our linters will report this once we add them to the project.

File: db/migrate/20200122143238_add_status_to_review_request.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToReviewRequest < ActiveRecord::Migration[6.0]
+  def change
+    add_column :review_requests, :status, :string
```
Comment (santib) [inline db/migrate/20200122143238_add_status_to_review_request.rb]: shouldn't we use an `enum` for statuses?

Comment (hdamico) [reply db/migrate/20200122143238_add_status_to_review_request.rb]: Done!

File: app/models/review_request.rb [hunk 7]
```diff
@@ -31,4 +31,11 @@ class ReviewRequest < ApplicationRecord
              foreign_key: :reviewer_id,
              inverse_of: :received_review_requests
   belongs_to :pull_request, inverse_of: :review_requests
+
+  enum status: { active: 'active', removed: 'removed' }
+  validates :status, inclusion: { in: statuses.keys }
+
+  def remove
```
Comment (santib) [inline app/models/review_request.rb]: you don't need this method, rails enums already gives you some handful method, for example in your case you want to do `pr.review_requests.find_by!(reviewer: reviewer)&.removed!`
